Dalamud/Game/Internal/Gui/DalamudIME.cs

```diff
@@ -0,0 +1,234 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using ImGuiNET;
+
+using static Dalamud.NativeFunctions;
+
+namespace Dalamud.Game.Internal.Gui
+{
+    /// <summary>
+    /// This class handles IME for non-English users.
+    /// </summary>
+    internal class DalamudIME : IDisposable
+    {
+        private static readonly ModuleLog Log = new("IME");
+
+        private readonly Dalamud dalamud;
+
+        private IntPtr interfaceHandle;
+        private IntPtr wndProcPtr;
+        private IntPtr oldWndProcPtr;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DalamudIME"/> class.
+        /// </summary>
+        /// <param name="dalamud">The Dalamud instance.</param>
+        internal DalamudIME(Dalamud dalamud)
+        {
+            this.dalamud = dalamud;
+        }
+
+        private delegate long WndProcDelegate(IntPtr hWnd, uint msg, ulong wParam, long lParam);
+
+        /// <summary>
+        /// Gets a value indicating whether the module is enabled.
+        /// </summary>
+        internal bool IsEnabled { get; private set; }
+
+        /// <summary>
+        /// Gets the imm candidates.
+        /// </summary>
+        internal List<string> ImmCand { get; private set; } = new();
+
+        /// <summary>
+        /// Gets the imm component.
+        /// </summary>
+        internal string ImmComp { get; private set; } = string.Empty;
+
+        /// <inheritdoc/>
+        public void Dispose()
+        {
+            if (this.oldWndProcPtr != IntPtr.Zero)
+            {
+                SetWindowLongPtrW(this.interfaceHandle, WindowLongType.WndProc, this.oldWndProcPtr);
+                this.oldWndProcPtr = IntPtr.Zero;
+            }
+        }
+
+        /// <summary>
+        /// Enables the IME module.
+        /// </summary>
+        internal void Enable()
+        {
+            try
+            {
+                this.interfaceHandle = this.dalamud.InterfaceManager.WindowHandlePtr;
+                this.wndProcPtr = Marshal.GetFunctionPointerForDelegate(new WndProcDelegate(this.WndProcDetour));
```

Comment:
My fix is like
- add a `WndProcDelegate` as a parameter of the class:
```
        private IntPtr interfaceHandle;
        private IntPtr wndProcPtr;
        private IntPtr oldWndProcPtr;
        private WndProcDelegate wndProcDelegate;
```
- set it to `WndProcDetour`:
```
        /// <summary>
        /// Enables the IME module.
        /// </summary>
        internal void Enable()
        {
            try
            {
                this.wndProcDelegate = this.WndProcDetour;
                this.interfaceHandle = this.dalamud.InterfaceManager.WindowHandlePtr;
                this.wndProcPtr = Marshal.GetFunctionPointerForDelegate(this.wndProcDelegate);
                this.oldWndProcPtr = SetWindowLongPtrW(this.interfaceHandle, WindowLongType.WndProc, this.wndProcPtr);
                this.IsEnabled = true;
                Log.Information("Enabled!");
            }
            catch (Exception ex)
            {
                Log.Information(ex, "Enable failed");
            }
        }
```